common/cpp/src/google_smart_card_common/value_conversion.h

```diff
@@ -174,7 +307,92 @@ class EnumValueDescriptor {
   internal::EnumFromValueConverter* const from_value_converter_;
 };
 
-///////////// ConvertToValue /////////////////
+///////////// StructValueDescriptor //////////////
+
+// Class that allows to describe a struct's fields and their corresponding
+// string names. Can be used in order to automatically implement conversion of a
+// C++ struct to/from a dictionary `Value` object (provided by the global
+// template functions `ConvertToValue()` and `ConvertFromValue()` defined
+// below).
+//
+// Example usage:
+//   struct Foo { int x; std::string y; };
+//   template <> StructValueDescriptor<Tp>::Description
+//   StructValueDescriptor<Tp>::GetDescription() {
+//     return Describe("Foo").WithField(&Foo::x, "x").WithField(&Foo::y, "y");
+//   }
+//   ...
+//   ConvertToValue(Foo{123, ""}, ...);
+//   ConvertFromValue(Value(...), &foo, ...);
+template <typename T>
+class StructValueDescriptor {
+ public:
+  static_assert(std::is_class<T>::value,
+                "The StructValueDescriptor parameter must be a class");
+
+  // Class that should be used for describing fields of the struct type. Should
+  // be instantiated via the `Describe()` method.
+  class Description final {
+   public:
+    Description(const char* type_name,
+                internal::StructToValueConverter<T>* to_value_converter,
+                internal::StructFromValueConverter<T>* from_value_converter)
+        : type_name_(type_name),
+          to_value_converter_(to_value_converter),
+          from_value_converter_(from_value_converter) {}
+    Description(Description&&) = default;
+    Description& operator=(Description&&) = default;
+    ~Description() = default;
+
+    const char* type_name() const { return type_name_; }
+
+    // Adds the given field into the struct's description: |dictionary_key_name|
+    // is the key in the dictionary `Value` representation of |field_ptr|.
+    // Returns a rvalue reference to |this| and uses the "&&" ref-qualifier, so
```

Comment:
nit: Returns **an** rvalue

Reply:
Done. Apologies, missed this after copy-pasting the unreviewed comment.